crates/holochain_conductor_api/src/config/conductor.rs

```diff
@@ -550,4 +695,126 @@ mod tests {
             }
         );
     }
+
+    #[test]
+    fn default_network_config_accepted_by_k2() {
+        let network_config = NetworkConfig::default();
+        let k2_config = network_config.to_k2_config().unwrap();
+
+        let builder = kitsune2_core::default_test_builder()
+            .with_default_config()
+            .unwrap();
+        builder.config.set_module_config(&k2_config).unwrap();
+        builder.validate_config().unwrap();
+    }
+
+    #[test]
+    fn network_config_preserves_advanced_overrides() {
+        let network_config = NetworkConfig {
+            advanced: Some(serde_json::json!({
+                "coreBootstrap": {
+                    "backoffMinMs": "3500",
+                },
+                "tx5Transport": {
+                    "timeoutS": "10",
+                },
+                "coreSpace": {
+                    "reSignFreqMs": "1000",
+                }
+            })),
+            ..Default::default()
+        };
+
+        let k2_config = network_config.to_k2_config().unwrap();
+
+        let builder = kitsune2_core::default_test_builder()
+            .with_default_config()
+            .unwrap();
+        builder.config.set_module_config(&k2_config).unwrap();
+        builder.validate_config().unwrap();
+        assert_eq!(
+            k2_config,
+            serde_json::json!({
+                "coreBootstrap": {
+                    "serverUrl": "https://devtest-bootstrap-1.holochain.org/",
+                    "backoffMinMs": "3500",
+                },
+                "tx5Transport": {
+                    "serverUrl": "wss://devtest-sbd-1.holochain.org/",
+                    "timeoutS": "10",
+                },
+                "coreSpace": {
+                    "reSignFreqMs": "1000",
+                }
+            })
+        )
+    }
+
+    #[test]
+    fn network_config_overrides_extracted_fields() {
```

Comment:
I'm not following this. Why should the builder change the config that you've applied to it? And why does it mutate the config you pass into it?

Reply:
This test is trying to suggest that if you set the bootstrap or signal URLs through the `advanced` field as well as the fields on the `NetworkConfig` then the ones on the `NetworkConfig` win. I think that is the behavior we want